### PR TITLE
[IoT] iot hub: Support IotHub in the profile of 2019-03-01-hybrid (#12069)

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Feature request: #12069

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S167

File: src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py [hunk 1]
```diff
@@ -414,9 +414,9 @@ def _get_eventhub_connectionstring(self, rg):
         return output.get_output_in_json()['primaryConnectionString']
 
     def _get_azurestorage_connectionstring(self, rg, container_name):
-        storage_name = 'iothubteststorage1'
+        storage_name = 'iothubteststorage11'
 
-        self.cmd('storage account create --resource-group {0} --name {1}'
+        self.cmd('storage account create --resource-group {0} --name {1} --https-only'
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py]: please use storage_account_preparer() for storage account create operation.

File: src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2019_03_01/test_iot_commands.py [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+                 .format(rg, ehNamespace, eventHub, eventHubPolicy, eventHubPolicyRight))
+
+        output = self.cmd('eventhubs eventhub authorization-rule keys list --resource-group {0} --namespace-name {1} --eventhub-name {2} --name {3}'
+                          .format(rg, ehNamespace, eventHub, eventHubPolicy))
+        return output.get_output_in_json()['primaryConnectionString']
+
+    def _get_azurestorage_connectionstring(self, rg, container_name):
+        storage_name = 'iothubteststorage2'
+
+        self.cmd('storage account create --resource-group {0} --name {1} --https-only'
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2019_03_01/test_iot_commands.py]: The same

File: src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py]: Done

File: src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2019_03_01/test_iot_commands.py (hunk 2 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2019_03_01/test_iot_commands.py]: Done

**Pull request merged**